src/response/status.rs

```diff
@@ -1,178 +1,492 @@
+    pub const fn as_warning_counter(self) -> Option<u8> {
+        if self.0 | Self::WARNING_COUNTER_MASK == Self::WARNING_COUNTER_MASK {
+            Some((self.0 | 0x00F) as u8)
+        } else {
+            None
+        }
+    }
 
-    // 69xx: command not allowed (cf. SW2)
-    SecurityStatusNotSatisfied,
-    ConditionsOfUseNotSatisfied,
-    OperationBlocked,
+    /// Create a warning counter status. (Meaning depends on the command)
+    ///
+    /// Value must be `0x00 <= value < 0x0F`, otherwise errors
+    pub const fn try_warning_counter(value: u8) -> Result<Self, WarningCounterError> {
+        if value <= 0xF {
+            Ok(Self(Self::WARNING_COUNTER_MASK | value as u16))
+        } else {
+            Err(WarningCounterError)
+        }
+    }
+    /// Value must be `0x00 <= value < 0x0F`, otherwise panics
+    pub const fn warning_counter(value: u8) -> Self {
+        match Self::try_warning_counter(value) {
+            Ok(s) => s,
+            Err(_) => panic!("Expected 0x00 <= value < 0x0F"),
+        }
+    }
+    pub const fn is_error_triggering(self) -> bool {
+        self.as_error_triggering().is_some()
+    }
+    pub const fn as_error_triggering(self) -> Option<u8> {
+        if matches!(
+            self.0,
+            Self::ERROR_TRIGGERING_LOWER..=Self::ERROR_TRIGGERING_UPPER
+        ) {
+            Some((self.0 & 0x00FF) as u8)
+        } else {
+            None
+        }
+    }
 
-    // 6Axx: wrong parameters P1-P2 (cf. SW2)
-    IncorrectDataParameter,
-    FunctionNotSupported,
-    NotFound,
-    NotEnoughMemory,
-    IncorrectP1OrP2Parameter,
-    KeyReferenceNotFound,
+    /// Value must be `0x02 <= value < 0x81`, otherwise panics
+    pub const fn error_triggering(value: u8) -> Self {
+        match Self::try_error_triggering(value) {
+            Ok(s) => s,
+            Err(_) => panic!("Expected 0x02 <= value < 0x81"),
+        }
+    }
+    /// Value must be `0x02 <= value < 0x81`, otherwise errors
+    pub const fn try_error_triggering(value: u8) -> Result<Self, TriggeringError> {
+        if value <= 0x80 && value >= 0x02 {
+            Ok(Self(Self::ERROR_TRIGGERING_LOWER | value as u16))
+        } else {
+            Err(TriggeringError)
+        }
+    }
 
-    // 6BXX: wrong parameters P1-P2
+    pub const fn as_wrong_le_field(self) -> Option<u8> {
+        if self.0 | Self::WRONG_LE_FIELD_MASK == Self::WRONG_LE_FIELD_MASK {
+            Some((self.0 & 0x00FF) as u8)
+        } else {
+            None
+        }
+    }
+    pub const fn is_wrong_le_field(self) -> bool {
+        self.as_wrong_le_field().is_some()
+    }
+    pub const fn as_bytes(self) -> [u8; 2] {
+        self.0.to_be_bytes()
+    }
 
-    // 6CXX: wrong Le field, SW2 encodes available bytes
+    pub const fn as_u16(self) -> u16 {
+        self.0
+    }
 
-    // 6D00: instruction code not supported or invalid
-    InstructionNotSupportedOrInvalid,
+    pub fn kind(self) -> Option<StatusKind> {
```

Comment:
In Rust we have TryInto for fallible conversions.

Reply:
In my mind this method is there to help with pattern matching, so you'd use it with:

```rust
match status.kind() {
    …
}
```

TryInto would require Turbofish and be much uglier. It can still be an option though.

I don't see what `must_use` would bring though.